zipline/finance/blotter/blotter_utils.py

```diff
@@ -0,0 +1,112 @@
+
+    Parameters
+    ----------
+    blotter_factories : dict[str -> function]
+        Factories for lazy blotter creation.
+    """
+    def __init__(self, blotter_factories):
+        self._blotter_factories = blotter_factories
+        self.blotter_factories = mappingproxy(self._blotter_factories)
+
+    def get_blotter_class(self, name):
+        """
+        Retrieves a blotter whose name is given.
+
+        Parameters
+        ----------
+        name : str
+            The name of the blotter class to be retrieved.
+
+        Returns
+        -------
+        blotter_class : zipline.finance.blotter.Blotter
+            The desired blotter class.
+        """
+        try:
+            return self._blotter_factories[name]
+        except KeyError:
+            raise ValueError("no blotter class registered as %r, options "
+                             "are: %r" % (name, sorted(self._blotter_factories
+                                                       )))
+
+    def has_blotter_class(self, name):
+        """
+        Whether or not the global list of blotter classes contains the
+        class with the specified name
+
+        :param name: The name of the blotter class
+
+        :return: True/False
+        """
+
+        return name in self._blotter_factories
+
+    def register_blotter_class(self, name, blotter_class=None):
+        """
+        Registers a blotter class for retrieval by the
+        get_blotter_class method
+
+        :param name: The name of the blotter class
+        :param blotter_class: The class to register, which must be a
+        subclass of the abstract class zipline.finance.blotter.Blotter
+        """
+
+        if blotter_class is None:
+            return partial(register_blotter_class, name)
+
+        if self.has_blotter_class(name):
+            raise ValueError("blotter class %r is already registered" % name)
+
+        if not issubclass(blotter_class, Blotter):
+            raise TypeError("The class specified is not a subclass of Blotter")
+
+        self._blotter_factories[name] = blotter_class
+
+        return blotter_class
+
+    def unregister_blotter_class(self, name):
+        """
+        If a blotter class is registered with the given name,
+        it is unregistered.
+
+        Parameters
+        ----------
+        name : str
+            The name of the blotter class to be unregistered.
+        """
+        try:
+            del self._blotter_factories[name]
+        except KeyError:
+            raise ValueError("blotter class %r was not already registered"
+                             % name)
+
+    def clear_blotter_classes(self):
+        """
+        Unregisters all current registered calendars
+        """
+        self._blotter_factories.clear()
+
+
+# Global blotter class dispatcher
+global_blotter_class_dispatcher = BlotterClassDispatcher(
+    blotter_factories={}
+)
+
+get_blotter_class = global_blotter_class_dispatcher.get_blotter_class
```

Comment:
sorry for all the churn on this, but could we rename these free functions and methods to drop the `_blotter_class` suffix. The expected use is something like:

```
from zipline import blotter


@blotter.register
class ...
```
in which case we have a prefix to disambiguate. This looks more like the other user extension points like bundles and metrics sets too.

Reply:
Ok, I'll do that